### PR TITLE
Fixed Me updating after EditCommand + Bug Fixes

### LINKED ISSUE #66: getFriendList() in ModelManager
It seems unnecessary for `getFriendList()` and `getOtherList()` in `ModelManager` to take in a `Person` parameter since the `User` is already declared in `ModelManager`.

### PR DESCRIPTION
**Pull request by cjinting-nus**
Changelog:
- Also fixed #66 , thanks Nian Fei for bringing it up, makes alot more sense like this
- Issue #175 is resolved
- Running group/find commands now return the correct number of persons
- friend/unfriend now tells you the person that is added/removed
- Changed tags back to only allow alphanumeric characters
- `edit` command does not recognise n/ prefixes anymore, tests are fixed to reflect that (#171 )
- undo/redo does not work with Friend/Unfriend, removed from the commandHistory

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FreeCommand.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public CommandResult execute(Model model, CommandHistory history) throws Command
         requireNonNull(model);
 
         User user = model.getUser();
```
Comment (nianfei97) [inline src/main/java/seedu/address/logic/commands/FreeCommand.java]: Probably remove this line too then

Comment (cjinting-nus) [reply src/main/java/seedu/address/logic/commands/FreeCommand.java]: You're right, thanks!

Comment (cjinting-nus) [reply src/main/java/seedu/address/logic/commands/FreeCommand.java]: Correction, I think it's still being used in the line after `DeconflictTimeTable deconflict = new DeconflictTimeTable(user.getTimeTable());`

Comment (nianfei97) [reply src/main/java/seedu/address/logic/commands/FreeCommand.java]: ah, alright. missed that one

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
 public class Tag {
 
     public static final String MESSAGE_CONSTRAINTS = "Tags names should be alphanumeric";
-    public static final String VALIDATION_REGEX = ".+";
+    public static final String VALIDATION_REGEX = "^[a-zA-Z0-9]+$";
```
Comment (cjinting-nus) [inline src/main/java/seedu/address/model/tag/Tag.java]: Corrected from #169

Comment (cjinting-nus) [reply src/main/java/seedu/address/model/tag/Tag.java]: Modified in latest commit to allow underscores and dashes

image

**Pull request merged**